src/inmanta/server/services/orchestrationservice.py

```diff
@@ -419,6 +427,151 @@ def safe_get(input: JsonType, key: str, default: object) -> object:
 
         return 200
 
+    @handle(methods.put_version, env="tid")
+    async def put_version(
+        self,
+        env: data.Environment,
+        version: int,
+        resources: List[JsonType],
+        resource_state: Dict[ResourceIdStr, const.ResourceState],
+        unknowns: List[Dict[str, PrimitiveTypes]],
+        version_info: model.ModelVersionInfo,
+        compiler_version: Optional[str] = None,
+        resource_sets: Optional[Dict[ResourceIdStr, Optional[str]]] = None,
+    ) -> Apireturn:
+        return await self._put_version(
+            env, version, resources, resource_state, unknowns, version_info, compiler_version, resource_sets
+        )
+
+    @handle(methods_v2.put_partial, env="tid")
+    async def put_partial(
+        self,
+        env: data.Environment,
+        version: int,
+        resources: list[Any],
+        resource_state: Dict[ResourceIdStr, ResourceState] = {},
+        unknowns: List[Dict[str, PrimitiveTypes]] = [],
+        version_info: Optional[model.ModelVersionInfo] = None,
+        compiler_version: Optional[str] = None,
+        resource_sets: Dict[ResourceIdStr, Optional[str]] = {},
+        removed_resource_sets: List[str] = [],
```

Comment:
I don't think the API layer verifies the `ResourceIdStr` type, because it's in reality just a alias for the str type (https://github.com/inmanta/inmanta-core/blob/6d247b20c7fd876febb1d01cff54ad312adc1303/src/inmanta/data/model.py#L181), but I could be wrong. Can you add a test to verify what happens when we provide a non resource id value as a key of one of those dictionaries?

Reply:
I wouldn't replace "ResourceIdStr" with "str". Mentioning it as "ResourceIdStr" has the advantage that it's clear to the reader that this should be a resource id. The only think that is missing here is the type validation. That still need to be added somehow. 

I would leave the test in. It would be a shame to throw away the effort we did in writing it. Next to that it performs an important check, so it's definitely useful.